sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1665,13 +1680,6 @@ def fit(self, X, y, sample_weight=None):
                 for l1_ratio in l1_ratios
             ]
         else:
-            # Making sure alphas entries are scalars.
-            if np.isscalar(alphas):
-                check_scalar_alpha(alphas, "alphas")
-            else:
-                # alphas is an iterable item in this case.
-                for index, alpha in enumerate(alphas):
-                    check_scalar_alpha(alpha, f"alphas[{index}]")
```

Comment:
We still need this part because `_validate_params` will only make sure that `alphas` is an array-like but we want to make sure that all its elements are >= 0.

Note that the branch above `if np.isscalar(alphas)` can be kept removed because it's unreachable code since alphas could never be a scalar.